### ISSUE TITLE
Feature Request: Create more robustness in configuration

### ISSUE THREAD

**Issue by MCKRUZ** [microsoft/Agents-for-net#15]
Currently when trying to use libraries like MSAL and other parts of the Agents framework it requires us to setup our appsettings.json a certain way with certain settings.  

One example of where this is done is when calling the "AddDefaultMsalAuth" method where you have to send in a configuration section for "MSALConfiguration".

Most of my enterprise level applications have custom configuration code where we inject various configurations from different locations and variations.  Rather then have the libraries pull directly from configuration could we possibly have an option to send in the variables directly?  I find myself having to write code to dynamically create configuration sections on the fly to do translations to just use the libraries.

**Comment by MCKRUZ** [comment]
Correct.  It's not only in the MSALConfiguration though it's all over the Agent framework.  For example I had to create my own version of the ConfigurationConnections class in order to make it so I didn't have to setup my appsettings.json in a certain manner.  That class takes in a IConfiguration object which directly pulls from the appsettings.  I try whenever possible to not recreate code that Microsoft has already built since then I have to watch for any changes you guys make and then make the appropriate changes on my side.  It would just make life easier if it were more robust to handle configuration outside of appsettings.json.

**Comment by tracyboehrer** [comment]
@MCKRUZ How did you handle IConnections?  Though that is why that is an interface, and different implementations could be created.

**Comment by tracyboehrer** [comment]
@MCKRUZ It would be interesting to see some examples.  It would be more accurate to say it is currently IConfiguration based, which may be things other than appsettings.  A programmatic way seems reasonable for many cases, and as we move on from "preview" status we'd like to support more flexible means.

**Comment by MCKRUZ** [comment]
So I ended up creating my own version of the IConnections class.  In my project I have a POCO object called AppConfig that handles our rather large config.  So to circumvent all of the places where you guys utilized IConfiguration, I simply created a helper class that loads the configs into memory and spits out a IConfigurationSection object whenever I need it.

**Comment by tracyboehrer** [comment]
@MCKRUZ This is still on my radar.  Not ignoring this.  Most of the newer stuff allows for code-first config instead of just IConfiguration.  Just haven't fit it into the schedule yet to make that work for the initial implementation of some things.  Namely MsalAuth and Connections.

**Comment by MCKRUZ** [comment]
No worries tracy, I know you guys are slammed.  I have a work around for now.  I did see you added AgentApplicationOptions, I'm digging into that functionality now.  Cheers mat.  Don't stress, you guys are killing it.
